File: evaluator.library/src/main/java/org/opencds/cqf/cql/evaluator/library/CqlParameterDefinition.java
package org.opencds.cqf.cql.evaluator.library;

public class CqlParameterDefinition {


    public CqlParameterDefinition(String name, String type, Boolean isList) {
        this(name, type, isList, null);
    }

    public CqlParameterDefinition(String name, String type, Boolean isList, Object value) {
        this.name = name;
        this.type = type;
        this.isList = isList;
        this.value = value;
    }

    private String name;
    private String type;
    private Object value;
    private Boolean isList;
    
    public String getName() {
        return this.name;
    }

    public String getType() {
        return this.type;
    }

    public Boolean getIsList() {
        return this.isList;
    }

    public Object getValue() {
        return this.value;
    }
    
}

File: cqf-fhir-cql/src/main/java/org/opencds/cqf/fhir/cql/engine/parameters/CqlParameterDefinition.kt
package org.opencds.cqf.fhir.cql.engine.parameters

import org.opencds.cqf.cql.engine.runtime.Value

class CqlParameterDefinition
@JvmOverloads
constructor(
    @JvmField val name: String,
    val type: String?,
    val isList: Boolean?,
    @JvmField val value: Value? = null,
)
